SHOWRAW.BAS

```diff
@@ -1,32 +1,116 @@
-REM A script to show the raw screen data saved in MANDELBR.BAS
-
 DECLARE SUB LoadRawScreen (filename AS STRING, w AS INTEGER, h AS INTEGER)
-
+DECLARE SUB CyclePalette ()
 SCREEN 13
 CLS
-LINE(0,0)-(320-200), 15, BF
+LINE (0, 0)-(320, 200), 15, BF
+CALL LoadRawScreen("C:\SCREEN.RAW", 320, 200)
 
-CALL LoadRawScreen("SCREEN.RAW", 320, 200)
+CALL CyclePalette
 
-SUB LoadRawScreen(filename AS STRING, w AS INTEGER, h AS INTEGER)
+SUB CyclePalette
+        DIM blue AS LONG
+        DIM green AS LONG
+        DIM red AS LONG
+        DIM pcolour AS LONG
+        DIM pvalues1(255) AS LONG
+        DIM pvalues2(255) AS LONG
+        DIM pvalues3(255) AS LONG
+        DIM pvalues4(255) AS LONG
+        FOR i = 0 TO 63
+                blue = i
+                green = i / 2
+                red = i / 3
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues1(i) = pcolour
```

Comment:
Looks like all of this could be reduced massively by a function that takes blue,green.red as parameters and returns the blue*65536+green*256+red.
Then you can just do
PCOLOUR(i, i/2, i/3) which nukes 4/5 lines.

Reply:
Problem with functions is that it requires a stack push, jump, code, then stack pop.
Unrolled it's faster. But I guess the speed doesn't matter massively here.

Reply:
Fixed